bio_files_processor.py

```diff
@@ -0,0 +1,120 @@
+                    seqs[name] = ''.join(seq)
+                name = line # initialize new name
+                seq = [] # reset sequence to empty list
+            else:
+                seq.append(line)
+        if name and seq: # add last sequence
+            seqs[name] = ''.join(seq)
+            
+    if not output_fasta:
+        output_fasta = os.path.splitext(input_fasta)[0] 
+
+    with open(f'{output_fasta}.fasta', 'w') as output_file:
+        for name, seq in seqs.items():
+            output_file.write(name + '\n')
+            output_file.write(seq + '\n')
+
+
+def get_genes_from_gbk(input_gbk: str) -> Dict[str, Tuple[str,str]]:
+    """
+    Extract genes and ther translation sequences from GBK file
+
+    Arguments:
+    - input_gbk(str): the path to the input GBK file 
+
+    Returns:
+    - genes_dict(Dict[str, Tuple[str,str]]): a dict with extracted genes
+    and their translation sequences in the format:
+    {gene: (order number, translation sequence)}
+    """
+    genes_dict = {}
+    gene_number = 0
+    index = 0
+    seqs = []
+    with open(input_gbk, 'r') as file:
+        lines = file.readlines()
+        while index < len(lines):
+            lines[index] = lines[index].strip()
+            if lines[index].startswith('/gene='): 
+                lines[index] = lines[index].strip('"\n')
+                gene_name = lines[index].replace('/gene="', '') 
+                index += 1
+                while not lines[index].strip().startswith('/translation='):  
+                    index += 1
+                while not lines[index].endswith('"\n'): #add multiline translation sequence
+                    seqs.append(lines[index].strip())
+                    index += 1
+                seqs.append(lines[index].strip().rstrip('"')) #add oneline or last line of multiline translation sequence
+                full_seq = ''.join(seqs).replace('/translation="', '')
+                seqs = []
+                genes_dict[gene_name] = gene_number, full_seq
+                gene_numner +=1
+            index += 1
+    return genes_dict
+
+
+def select_genes_from_gbk_to_fasta(input_gbk: str, genes: list, 
+                                   n_before: int = 1, n_after: int = 1,
+                                   output_fasta: str = None) -> None:
+    """
+    Select genes from GBK file that are adjacent to the genes of interest
+
+    Arguments:
+    - input_gbk (str): the path to the input GBK file 
+    - genes (list): a list of names to genes of interest
+    - n_before (int, default = 1): number of genes to extract before the gene of interest
+    - n_after (int, default = 1): number of genes to extract after the gene of interest
+    - output_fasta (str, optional): name of the output FASTA file 
+    The default output file name is the name of the input file.
+    """
+
+    genes_dict = get_genes_from_gbk(input_gbk)
+    genes_from_gbk = list(genes_dict.keys()) 
+    max_number = len(genes_only)
```

Comment:
```suggestion
    max_number = len(genes_from_gbk)
```
из за этой ошибки все не работает и падает, снова очень обидно:)